components/camel-csv/src/main/java/org/apache/camel/dataformat/csv/CsvMarshaller.java

```diff
@@ -85,6 +85,17 @@ public void marshal(Exchange exchange, Object object, OutputStream outputStream)
         }
     }
 
+    /**
+     * Creates and returns a {@link CSVPrinter}.
+     *
+     * @param exchange     Exchange (used for access to type conversion). Could NOT be <code>null</code>.
+     * @param outputStream Output stream of the CSV. Could NOT be <code>null</code>.

```

Comment:
You can use objecthelper.notnull to enforce non-null values

Reply:
Thanks for the hint. There is `java.util.Objects.requireNonNull` since **Java 1.7** as well. Personally, I would prefer _assertions_ for potential programmatic errors like here (_if the value is null, then the programmer did something wrong - not related to the environment or to the machine_).
However, I am going to follow your suggestion (for consistency) and adapt the PR.